runtime/compiler/optimizer/J9TransformUtil.cpp

```diff
@@ -1982,3 +1982,63 @@ void J9::TransformUtil::prohibitOSROverRange(TR::Compilation* comp, TR::TreeTop*
       }
    }
 
+static void prohibitGuardedStaticFinalFieldFoldingOnNodeAndChildren(TR::Compilation* comp, TR::Node* node, TR::NodeChecklist *visited)
+   {
+   if (visited->contains(node))
+      return;
+
+   visited->add(node);
+
+   int32_t numChildren = node->getNumChildren();
+   if (numChildren == 0 &&
+       node->getOpCode().isLoadVarDirect() &&
+       node->isLoadOfStaticFinalField())
+      {
+      traceMsg(comp, "prohibit folding on n%dn\n", node->getGlobalIndex());
```

Comment:
There should be a some kind of trace flag or something so this doesn't fill the log up.

Reply:
This is where we set the flag, so I updated the code to use dumpOptDetails here.